Neos.Neos/Classes/Utility/NodeUriPathSegmentGenerator.php

```diff
@@ -38,19 +41,36 @@ class NodeUriPathSegmentGenerator
      */
     public function generateUriPathSegment(?Node $node = null, ?string $text = null): string
     {
-        $language = null;
-        if ($node) {
-            $text = $text ?: $node->getLabel() ?: ($node->nodeName?->value ?? '');
-            $languageDimensionValue = $node->originDimensionSpacePoint->coordinates['language'] ?? null;
-            if (!is_null($languageDimensionValue)) {
-                $locale = new Locale($languageDimensionValue);
-                $language = $locale->getLanguage();
-            }
-        } elseif (is_null($text) || empty($text)) {
+        if ($node === null && empty($text)) {
             throw new Exception('Given text was empty.', 1457591815);
         }
-        $text = $this->transliterationService->transliterate($text, $language);
 
-        return Transliterator::urlize($text);
+        $textForNode = $text ?: $node->getLabel() ?: $node->nodeName?->value ?? '';
+
+        return $this->generateUriPathSegmentFromTextForDimension(
+            $textForNode,
+            $node?->originDimensionSpacePoint->toDimensionSpacePoint() ?? DimensionSpacePoint::fromArray([])
+        );
+    }
+
+    /**
+     * Generates a URI path segment for a given text taking the language dimension value into account
+     *
+     * @param string $text
+     * @param DimensionSpacePoint $dimensionSpacePoint to determine language dimension value from
+     */
+    public function generateUriPathSegmentFromTextForDimension(string $text, DimensionSpacePoint $dimensionSpacePoint): string
+    {
+        $languageDimensionValue = $dimensionSpacePoint->getCoordinate(new ContentDimensionId('language'));
+        $language = null;
+        if ($languageDimensionValue !== null) {
+            try {
+                $language = (new Locale($languageDimensionValue))->getLanguage();
+            } catch (InvalidLocaleIdentifierException $e) {
+                // we don't need to do anything here; we'll just transliterate the text.
```

Comment:
This is new it seems and I always get skeptical if exceptions are just swallowed..
Why do we need it now?

Reply:
Hmm yes idk, YOU @bwaidelich  tell me ;)

commit:
https://github.com/neos/contentrepository-development-collection/commit/2e80b9f7d1336cf936d2d79946e96f71577f726a#diff-0072c01664350f7d6780a510817e4633a4fc538fe566eb384eadc65176d5556dR72


line in Neos.EventSourcedNeosAdjustments 
https://github.com/neos/contentrepository-development-collection/blame/693e65edb54b82a173cb925f72c2b76bd6ff4efb/Neos.EventSourcedNeosAdjustments/Classes/Ui/NodeCreationHandler/DocumentTitleNodeCreationHandler.php#L93

line in neos ui
https://github.com/neos/neos-ui/blame/9585ca792616d52963e69f71c12c1915dfd5cea4/Classes/NodeCreationHandler/DocumentTitleNodeCreationHandler.php#L88

Reply:
Thehe, YOU @skurfuerst tell me => https://github.com/neos/contentrepository-development-collection/commit/17b7e2f8267b3e66c00db4a60049c66d122eea93#diff-ecc5b686b24c260fc64480dfac267e6fca721c028038c1a5221a5c752420937eR52

Reply:
i think that catching the exception is fine for the following reason:

We try to automagically infer the locale from the language dimension of the dsp. But the dsp is doesnt enforce a specific locale format. For what its worth one could write `language => foo-bar432423` which would be a valid dimension configuration. As we only try our best to get the language based on the dsp i think its okay to not throw an exception (that the caller possibly has to worry about) as nothing really breaks - we will just not translate it properly.

In case you want your own setup to enforce that the language dimension is a valid locale, one should write a test or validate command idk ^^

Reply:
From my POV it's fine to catch it here, as this is just guessing. And if we find something, we can improve the guess. Otherwise the fallback kicks in.